docarray/array/mixins/io/pushpull.py

```diff
@@ -126,21 +159,34 @@ def pull(
         """Pulling a :class:`DocumentArray` from Jina Cloud Service to local.
 
         :param token: the upload token set during :meth:`.push`
+        :param id: the unique identifier that returned from :meth:`.push`
         :param show_progress: if to show a progress bar on pulling
         :param local_cache: store the downloaded DocumentArray to local folder
         :return: a :class:`DocumentArray` object
         """
 
         import requests
 
-        url = f'{_get_cloud_api()}/v2/rpc/da.pull?token={token}'
-        response = requests.get(url)
+        headers = {}
+
+        auth_token = _get_hub_config().get('auth_token')
+        if auth_token:
+            headers['Authorization'] = f'token {auth_token}'
+
+        params = {}
+        if token:
+            params['token'] = token
+        if id:
+            params['id'] = id
+
+        url = f'{_get_cloud_api()}/v2/rpc/da.pull?{parse.urlencode(params)}'
+        response = requests.get(url, headers=headers)
 
         if response.ok:
             url = response.json()['data']['download']
         else:
             raise FileNotFoundError(
```

Comment:
The error here is misleading, maybe we should raise something like `BadRequestError` or so ?